### PR TITLE
doc: Add Code of Conduct

### PR DESCRIPTION
**Pull request by tgraf**
We'll use the CNCF Code of Conduct as it is well written, well-known an
appropriate for Cilium.

### CODE REVIEW

File: CODE_OF_CONDUCT.md [hunk 1]
```diff
@@ -0,0 +1,3 @@
+# Community Code of Conduct
+
+Cilium follows the [CNCF Code of Conduct](https://github.com/cncf/foundation/blob/master/code-of-conduct.md).
```
Comment (aanm) [inline CODE_OF_CONDUCT.md]: at the risk of the link become 404, shouldn't we copy paste the entire text into cilium?

**Review by joestringer — APPROVED** [review]
I'm good with this in principle. I note that if we do point to a common CoC from another community then that means we will automatically adopt any future changes without otherwise having any notification mechanism. I don't predict any particular issues with this, but it's the other side of the coin to Andre's feedback.

File: CODE_OF_CONDUCT.md (hunk 1 above)
Comment (joestringer) [inline CODE_OF_CONDUCT.md]: Even if we follow this code of conduct, we should make it clear how conduct violations should be reported as the upstream document here refers to either conduct@kubernetes.io (for kubernetes, so doesn't apply here), "A CNCF project maintainer" which is probably the closest but doesn't define any method, or the CNCF mediator Mishi Choudhary.

Comment (tgraf) [reply CODE_OF_CONDUCT.md]: All good points, I've included it and adjusted contact details.

Comment (tgraf) [reply CODE_OF_CONDUCT.md]: I've modified it

**Review by jrajahalme — APPROVED** [review]

**Pull request merged**